### PR TITLE
Autocalculate ALLOWED_NOTREADY_NODES based on NUM_NODES

### PR DESCRIPTION
**Pull request by shyamjvs**
As an alternative for https://github.com/kubernetes/test-infra/pull/7698

/cc @wojtek-t

```release-note
NONE
```

### CODE REVIEW

File: cluster/validate-cluster.sh [hunk 1]
```diff
@@ -47,7 +47,9 @@ function kubectl_retry() {
   done
 }
 
-ALLOWED_NOTREADY_NODES="${ALLOWED_NOTREADY_NODES:-0}"
+# To avoid failing large tests due to some flakes in starting nodes, allow
+# for a small percentage of nodes to not start during cluster startup.
+ALLOWED_NOTREADY_NODES="${ALLOWED_NOTREADY_NODES:-$((NUM_NODES / 100))}"
```
Comment (wojtek-t) [inline cluster/validate-cluster.sh]: This has two drawback compared to https://github.com/kubernetes/test-infra/pull/7698#discussion_r181684628
1. this function is not used only in tests, and outside of tests people not necessary expect such behavior
2. I don't want to cherrypick it :)

Comment (shyamjvs) [reply cluster/validate-cluster.sh]: The first one I can agree with - PTAL now (it's not affecting prod).
The second one I'm not so sure about (i.e not choosing a fix because it needs a cherrypick):
- we're choosing test fixes into release branches from time to time?
- the only release branch scalability jobs we're running are 100-node ones (and I'm not sure if this issue (i.e 1 unready node) is happening frequently enough to care about cherry-picking)

Comment (wojtek-t) [reply cluster/validate-cluster.sh]: I still think that we shouldn't change that in upstream.
It's a setting that we would like to make in our tests, but not necessary impose at upstream.
So I still think change in test-infra is better.

[Re cherrypick - sure, that shouldn't be determining factor]

Comment (wojtek-t) [reply cluster/validate-cluster.sh]: > That said, if you still want to test large clusters (which I'm not aware anyone except us are doing) with 0 unready nodes, you can override it.

Zero people know about this env var. I don't think changing that default is a good way to proceed.

> Test-infra code already has quite some complexity (adding more such dependencies in that big main.go file will just make things more harder to follow/maintain IMO).

Still waaay simpler that our setup scripts, which are freaking hard to follow...

Comment (shyamjvs) [reply cluster/validate-cluster.sh]: That said, I wouldn't push super hard on this. It just feels more cleaner to me to have this one-liner change in our test scripts here (we're already doing some large-cluster custom things in our scripts here).

Comment (shyamjvs) [reply cluster/validate-cluster.sh]: > Zero people know about this env var. I don't think changing that default is a good way to proceed.

If that's the problem, we can explicitly log this as part of cluster-validation step (we're already printing the no. of expected nodes there). Does that SGTY?

> Still waaay simpler that our setup scripts, which are freaking hard to follow...

This one is hard to argue I think :)
(but in general, it takes me lesser time to understand and make changes in our k/k setup scripts compared to test-infra code (which also keeps changing all the time wrt flags/env-vars))

Comment (wojtek-t) [reply cluster/validate-cluster.sh]: > If that's the problem, we can explicitly log this as part of cluster-validation step (we're already printing the no. of expected nodes there). Does that SGTY?

Adding a log within a long stream of logs that our startup scripts produce doesn't solve the problem, if you don't know what to look for. It helps only those who know what they are looking for which is not the case here...

> (but in general, it takes me lesser time to understand and make changes in our k/k setup scripts compared to test-infra code (which also keeps changing all the time wrt flags/env-vars))

Interesting - I have exactly the opposite feeling :)

Comment (shyamjvs) [reply cluster/validate-cluster.sh]: Ok, I've moved it now to config-test.sh which is "the" file having all our test configs. This is the same file NUM_NODES (and many other key variables) are defined.
I would expect someone who wants to test a large cluster on GCE to at least look at this file :)

Does it LGTY?

Comment (wojtek-t) [reply cluster/validate-cluster.sh]: I'm still not convinced this is actually better than changes in test-infra. I need to think about it :)

Comment (shyamjvs) [reply cluster/validate-cluster.sh]: Also, I think we should make such parameters for creating large clusters open and clearly visible to people (instead of silently doing it within our tests). Other examples of that (in the same file) are adjusting master-size, master disk size, etc.. Similarly, we should also make visible about this long tail latency to expect when creating large clusters.
